### PR TITLE
feat(wallet): adds wallet feature using xpubs

### PR DESCRIPTION
**Pull request by bucko13**
Branch with ongoing development for new wallet update

### CODE REVIEW

**Comment by waldenraines** [comment]
> * Can we pick one of either `.js` or .`jsx` as file endings?

Currently we are using `.jsx` for files that contain `jsx` and `.js` for files that don't.  This comes from [this eslint rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-filename-extension.md).  

I suppose we could use `.jsx` for everything but that seems a little strange to me.  The other option would be to disable this rule.

**Comment by bucko13** [comment]
> Currently we are using .jsx for files that contain jsx and .js for files that don't. This comes from this eslint rule.
> I suppose we could use .jsx for everything but that seems a little strange to me. The other option would be to disable this rule.

This unfortunately isn't true at the moment. You can look at the reducers as an example of files with no JSX but still have the extension. I'm in favor of either changing all but components (i.e. files with JSX) to the `.jsx` extension or all `.js` if there are no compile issues.

**Comment by waldenraines** [comment]
> This unfortunately isn't true at the moment. You can look at the reducers as an example of files with no JSX but still have the extension. I'm in favor of either changing all but components (i.e. files with JSX) to the `.jsx` extension or all `.js` if there are no compile issues.

Oh.  I didn't realize there were existing files that used `.jsx` before my big linting fixes.  Yes, I agree that reducers should not be using the `.jsx` extension :facepalm:.

File: src/reducers/addressReducer.jsx [hunk 1]
```diff
@@ -21,81 +15,106 @@ import {
...
-    ...{publicKeyImporters: publicKeyImportersChange},
+    ...{ publicKeyImporters: publicKeyImportersChange },
   };
-  return updateState(newState, {fingerprint: fingerprint(newState)});
+  return updateState(newState, { fingerprint: fingerprint(newState) });
+}
+
+function setConflict(publicKeyImporter, state) {
+  if (state.finalizedNetwork) {
+    // eslint-disable-next-line no-param-reassign
```
[the remark this answers is not in the document]
Comment (bucko13) [reply src/reducers/addressReducer.jsx:125]: No idea

**Pull request merged**